addon/formatters/-base.js

```diff
@@ -65,7 +65,7 @@ const FormatBase = Ember.Object.extend({
 });
 
 FormatBase.reopenClass({
```

Comment:
I think we should handle this at the helper level and not the formatter.  Basically don't invoke the formatters if value is empty when allowEmpty is truthy